examples/demos/create_infrastructure_with_storage.pp

```diff
@@ -0,0 +1,288 @@
+    poolName         => 'CPG-SSD-AO',
+    storageSystemUri => 'ThreePAR-1'
+  }
+}
+
+# Creates Storage Volume in OneView
+oneview_volume{'Volume':
+    ensure  => 'present',
+    require => Oneview_storage_pool['Storage Pool'],
+    data    => {
+      properties  => {
+            provisioningType => 'Thin',
+            size             => 1073741824,
+            name             => 'Oneview_Puppet_TEST_VOLUME_1',
+            storagePool      => '/rest/storage-pools/C6190E80-7246-42BF-92B8-AB9E00CFF1C6',
+            snapshotPool     => '/rest/storage-pools/C6190E80-7246-42BF-92B8-AB9E00CFF1C6',
+            isShareable      => false
+      }
+    }
+}
+
+$interconnect_type_1 = 'Virtual Connect SE 40Gb F8 Module for Synergy'
+$interconnect_type_2 = 'Synergy 20Gb Interconnect Link Module'
+$network_names = [ 'Test Puppet Network' ]
+
+# Creates LIG with uplinkSets, Redundancy, Boot settings in Synergy
+oneview_logical_interconnect_group{'Test Puppet LIG':
+  ensure  => 'present',
+  require => Oneview_ethernet_network['Test Puppet Network'],
+  data    => {
+    name               => 'Test Puppet LIG',
+    redundancyType     => 'HighlyAvailable',
+    interconnectBaySet => 3,
+    enclosureType      => 'SY12000',
+    enclosureIndexes   => [1, 2, 3],
+    uplinkSets         =>
+    [
+      {
+        name                => 'TUNNEL_ETH_UP_01',
+        ethernetNetworkType => 'Tagged',
+        networkType         => 'Ethernet',
+        lacpTimer           => 'Short',
+        mode                => 'Auto',
+        uplink_ports        => [{ bay             => 3,
+                                  port            => 'Q1',
+                                  type            => $interconnect_type_1,
+                                  enclosure_index => 1 },
+                                { bay             => 6,
+                                  port            => 'Q1',
+                                  type            => $interconnect_type_1,
+                                  enclosure_index => 2 }
+        ],
+        networkUris         => $network_names
+      }
+    ],
+    interconnects      =>
+    [
+      {
+        bay             => 3,
+        type            => $interconnect_type_1,
+        enclosure_index => 1
+      },
+      {
+        bay             => 6,
+        type            => $interconnect_type_1,
+        enclosure_index => 2
+      },
+      {
+        bay             => 3,
+        type            => $interconnect_type_2,
+        enclosure_index => 2
+      },
+      {
+        bay             => 6,
+        type            => $interconnect_type_2,
+        enclosure_index => 1
+      },
+      {
+        bay             => 3,
+        type            => $interconnect_type_2,
+        enclosure_index => 3
+      },
+      {
+        bay             => 6,
+        type            => $interconnect_type_2,
+        enclosure_index => 3
+      }
+    ]
+  }
+}
+
+# Creates EnclosureGroup using LIG created above
```

Comment:
Do not abbreviate resource names in comments

Reply:
done